vraptor-core/src/main/java/br/com/caelum/vraptor/observer/InstantiateObserver.java

```diff
@@ -58,7 +52,7 @@ public void instantiate(@Observes ControllerFound event) {
 		this.controllerInstance = new DefaultControllerInstance(instance);
 	}
 
-	@Produces @RequestScoped
+	@Produces @Dependent
 	public ControllerInstance getControllerInstance() {
```

Comment:
this bean observes `ControllerFound` updating the controller instance, but since it was declared as request scoped, after a forward even firing the event, CDI was caching the old/first controller instance in the request scope.